### DISCUSSION TITLE
add another language option?

### DISCUSSION THREAD

**Discussion by LDalby** [rstudio/pointblank#302] [General]
Hi Rich!

I've recently started using this package and it is amazing! So thanks a lot 🙏🏼 

Today I discovered the cool options for different languages in the reports. I can see this being very useful in my use cases, but - you guessed it - my native language is not one of the options. 

So I was wandering if you'd consider a PR to add translations to Danish?

**Comment by rich-iannone** [comment]
I would love this, thank you! This would be incredibly valuable so let me help you get started:

1. Modify this file https://github.com/rich-iannone/pointblank/blob/master/data-raw/translations_source.yml by adding a `da: ` entry to each block of strings.
2. Then, this vector must be extended to include `"da"`: https://github.com/rich-iannone/pointblank/blob/95f396947175f9941498e320458a888425b3810c/R/utils_translations.R#L21
3. To see the translation actually work, you need to source the file https://github.com/rich-iannone/pointblank/blob/master/data-raw/00-reencode_text.R to rebuild the internal dataset that has all of the translations. Then rebuild the package, and run `scan_data("small_table", lang = "da")`. 

Then the rest is just updating the docs (which I can do, if you like). In fact the biggest part will be step (1), I can jump in and help at any time to assist, just ask.

Thanks again, this will be a great addition!

**Reply by LDalby** [reply]
Great! 

Thanks for the pointers, I'll try to get this going as soon as I can. 

I'll let you know if I get stuck or need help otherwise.

**Reply by rich-iannone** [reply]
Thanks! And reach out anytime.

**Reply by LDalby** [reply]
Right, so the steps 1-3 done now. 

I get this in 3) : 

```r
> source('~/git/pointblank/data-raw/00-reencode_text.R')
── Attaching packages ───────────────────────────────────────────────────────────────────────────────── tidyverse 1.3.0.9000 ──
✓ ggplot2 3.3.3          ✓ purrr   0.3.4     
✓ tibble  3.1.0          ✓ dplyr   1.0.5     
✓ tidyr   1.1.3          ✓ stringr 1.4.0     
✓ readr   1.4.0.9000     ✓ forcats 0.5.1.9000
── Conflicts ───────────────────────────────────────────────────────────────────────────────────────── tidyverse_conflicts() ──
x dplyr::filter() masks stats::filter()
x dplyr::lag()    masks stats::lag()
here() starts at /home/au206907/git/pointblank
There were 50 or more warnings (use warnings() to see the first 50)
> warnings()
Warning messages:
1: In stringi::stri_unescape_unicode(.) :
  invalid escape sequence detected; setting NA
2: In stringi::stri_unescape_unicode(.) :
  invalid escape sequence detected; setting NA
3: In stringi::stri_unescape_unicode(.) :
  invalid escape sequence detected; setting NA
# continues to 50
  ```
and this error when trying to scan:

``` r
library(pointblank); scan_data("small_table", lang = "da")
```

    #> 

    #> ── Data Scan started. Processing 6 sections. ───────────────────────────────────

    #> ℹ Starting assembly of 'Overview' section...

    #> Error in UseMethod("count"): no applicable method for 'count' applied to an object of class "character"

Do you have any idea what I could be doing wrong here?


<sup>Created on 2021-03-18 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0.9002)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
```

    #> ─ Session info ───────────────────────────────────────────────────────────────
    #>  setting  value                       
    #>  version  R version 4.0.4 (2021-02-15)
    #>  os       Ubuntu 20.04.2 LTS          
    #>  system   x86_64, linux-gnu           
    #>  ui       X11                         
    #>  language en_US:en                    
    #>  collate  en_US.UTF-8                 
    #>  ctype    en_US.UTF-8                 
    #>  tz       Europe/Copenhagen           
    #>  date     2021-03-18                  
    #> 
    #> ─ Packages ───────────────────────────────────────────────────────────────────
    #>  package     * version    date       lib source        
    #>  assertthat    0.2.1      2019-03-21 [3] CRAN (R 4.0.0)
    #>  blastula      0.3.2      2021-02-24 [1] local         
    #>  cli           2.3.1.9000 2021-03-01 [1] local         
    #>  colorspace    2.0-0      2020-11-11 [3] CRAN (R 4.0.3)
    #>  crayon        1.4.1      2021-02-08 [3] CRAN (R 4.0.3)
    #>  DBI           1.1.1      2021-01-15 [1] CRAN (R 4.0.4)
    #>  digest        0.6.27     2020-10-24 [3] CRAN (R 4.0.3)
    #>  dplyr         1.0.5      2021-02-24 [1] local         
    #>  ellipsis      0.3.1      2020-05-15 [3] CRAN (R 4.0.0)
    #>  evaluate      0.14       2019-05-28 [3] CRAN (R 4.0.0)
    #>  fansi         0.4.2      2021-01-15 [3] CRAN (R 4.0.3)
    #>  farver        2.1.0      2021-02-28 [3] CRAN (R 4.0.4)
    #>  fs            1.5.0.9000 2021-02-24 [1] local         
    #>  generics      0.1.0.9000 2021-02-24 [1] local         
    #>  ggforce       0.3.3      2021-03-05 [1] CRAN (R 4.0.4)
    #>  ggplot2       3.3.3      2020-12-30 [3] CRAN (R 4.0.3)
    #>  glue          1.4.2      2020-08-27 [3] CRAN (R 4.0.2)
    #>  gtable        0.3.0      2019-03-25 [3] CRAN (R 4.0.0)
    #>  highr         0.8        2019-03-20 [3] CRAN (R 4.0.0)
    #>  htmltools     0.5.1.1    2021-01-22 [3] CRAN (R 4.0.3)
    #>  knitr         1.31       2021-01-27 [3] CRAN (R 4.0.3)
    #>  lifecycle     1.0.0      2021-02-15 [3] CRAN (R 4.0.4)
    #>  magrittr      2.0.1.9000 2021-02-24 [1] local         
    #>  MASS          7.3-53.1   2021-02-12 [4] CRAN (R 4.0.3)
    #>  munsell       0.5.0      2018-06-12 [3] CRAN (R 4.0.0)
    #>  pillar        1.5.1      2021-03-05 [3] CRAN (R 4.0.4)
    #>  pkgconfig     2.0.3      2019-09-22 [3] CRAN (R 4.0.0)
    #>  pointblank  * 0.7.0.9000 2021-03-18 [1] local         
    #>  polyclip      1.10-0.001 2021-02-24 [1] local         
    #>  purrr         0.3.4      2020-04-17 [3] CRAN (R 4.0.0)
    #>  R6            2.5.0      2020-10-28 [3] CRAN (R 4.0.3)
    #>  Rcpp          1.0.6      2021-01-15 [3] CRAN (R 4.0.3)
    #>  reprex        1.0.0.9002 2021-02-24 [1] local         
    #>  rlang         0.4.10     2020-12-30 [3] CRAN (R 4.0.3)
    #>  rmarkdown     2.7        2021-02-19 [3] CRAN (R 4.0.4)
    #>  rstudioapi    0.13       2020-11-12 [3] CRAN (R 4.0.3)
    #>  scales        1.1.1      2020-05-11 [3] CRAN (R 4.0.0)
    #>  sessioninfo   1.1.1      2018-11-05 [3] CRAN (R 4.0.0)
    #>  stringi       1.5.3      2020-09-09 [3] CRAN (R 4.0.2)
    #>  stringr       1.4.0      2019-02-10 [3] CRAN (R 4.0.0)
    #>  tibble        3.1.0      2021-02-25 [3] CRAN (R 4.0.4)
    #>  tidyselect    1.1.0.9000 2021-02-24 [1] local         
    #>  tweenr        1.0.1      2018-12-14 [1] CRAN (R 4.0.4)
    #>  utf8          1.2.1      2021-03-12 [3] CRAN (R 4.0.4)
    #>  vctrs         0.3.6      2020-12-17 [3] CRAN (R 4.0.3)
    #>  withr         2.4.1      2021-01-26 [3] CRAN (R 4.0.3)
    #>  xfun          0.22       2021-03-11 [3] CRAN (R 4.0.4)
    #>  yaml          2.2.1      2020-02-01 [3] CRAN (R 4.0.0)
    #> 
    #> [1] /home/au206907/R/x86_64-pc-linux-gnu-library/4.0
    #> [2] /usr/local/lib/R/site-library
    #> [3] /usr/lib/R/site-library
    #> [4] /usr/lib/R/library

</details>

**Reply by rich-iannone** [reply]
It seems like there are some issues with the translation strings. If you're this far ahead, could you create the PR? That way I can test it on my system and debug the problem (I'm sure it's a very small set of problems).

**Reply by LDalby** [reply]
I've opened the PR just now.

**Reply by rich-iannone** [reply]
It’s all merged in now. Thanks again for putting in all the time here, it’s really a lot of strings.

**Reply by LDalby** [reply]
Excellent, thanks! 
Agreed, quite a lot of strings 😄
